plugins/scaffolder-node/src/actions/executeShellCommand.ts

```diff
@@ -45,9 +48,24 @@ export async function executeShellCommand(
     command,
     args,
     options: spawnOptions,
-    logStream = new PassThrough(),
+    logger,
+    logLevel = 'verbose',
   } = options;
 
+  // Writable stream that logs stdout and stderr using the winston logger and user specified log level
+  const logStream = new Writable({
```

Comment:
I think we can skip creating one by default, and only write to it if one is provided. This would mean that the logic in the below section:

https://github.com/backstage/backstage/blob/7abee9ae662cb9415e6b4496d88592b8dae6b0d6/plugins/scaffolder-node/src/actions/executeShellCommand.ts#L72-L78

Becomes something like:

```ts
 process.stdout.on('data', stream => { 
   logStream?.write(stream);
   logger?.log(Buffer.isBuffer(chunk)
        ? chunk.toString('utf8').trim()
        : chunk.toString(encoding || 'utf8').trim());
 }); 
  
 process.stderr.on('data', stream => { 
   logStream?.write(stream);
   logger?.error(Buffer.isBuffer(chunk)
        ? chunk.toString('utf8').trim()
        : chunk.toString(encoding || 'utf8').trim());
 }); 
```

Maybe need a little cleanup for the parsing of the message of course :)

Also note how we're removing the need for the `logLevel` at all, are we're setting it depending on if theres an error or not from the original process stream.

Reply:
Hi @benjdlambert ,

Thanks a lot for your input! 
I left this PR as a draft because I wasn't happy with the solution and also because I'm on vacation. I'm be back in 2 weeks. Is it ok if I get back to it in the first week of September?  😸

Reply:
No probs! Have a great vacation!